src/main/java/dev/cammiescorner/arcanuscontinuum/common/blocks/MagicDoorBlock.java

```diff
@@ -53,17 +57,23 @@ public void scheduledTick(BlockState state, ServerWorld world, BlockPos pos, Ran
 
 	@Override
 	public ActionResult onUse(BlockState state, World world, BlockPos pos, PlayerEntity player, Hand hand, BlockHitResult hit) {
+		if (world.isClient) return ActionResult.SUCCESS;
+
 		ItemStack stack = player.getStackInHand(hand);
 		MagicDoorBlockEntity door = getBlockEntity(world, state, pos);
+		LivingEntity owner = door.getOwner();
 
-		if(stack.isOf(Items.NAME_TAG) && stack.hasCustomName() && door != null)
-			if(door.getOwner().equals(player))
-				door.setPassword(stack.getName().getString());
-			else
+		if(owner != null && stack.isOf(Items.NAME_TAG) && stack.hasCustomName())
+			if(owner.getUuid().compareTo(player.getUuid()) == 0) {
```

Comment:
any reason for using `compareTo()` instead of `equals()`?

Reply:
Nope lol my mistake